app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ReferCaseController.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc._
+import play.twirl.api.HtmlFormat
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.models.{Case, CaseStatus}
+import uk.gov.hmrc.tariffclassificationfrontend.service.CasesService
+import uk.gov.hmrc.tariffclassificationfrontend.views
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.Future.successful
+
+@Singleton
+class ReferCaseController @Inject()(authenticatedAction: AuthenticatedAction,
```

Comment:
This controller looks very similar to the `ReleaseCaseController` and the `CompleteCaseController`.
Where possible, would be nice to extract the common code in a parent trait and re-use such common code in all these controllers.

Reply:
Maybe in the next story, once we write reOpen case

Reply:
To me they are all different actions though. We should avoid inheritance if possible